client/src/app/project-registration/project-registration-form/project-registration-form.component.ts

```diff
@@ -44,17 +47,41 @@ export class ProjectRegistrationFormComponent implements OnInit {
 
   @ViewChild('mf') formTabGroup: MatTabGroup;
 
+  private emptyProject = {content: {}};
+
   constructor(private route: ActivatedRoute,
               private router: Router,
               private ingestService: IngestService,
               private alertService: AlertService,
               private loaderService: LoaderService,
-              private schemaService: SchemaService) {
+              private schemaService: SchemaService,
+              private autofillProjectService: AutofillProjectService,
+  ) {
   }
 
   ngOnInit() {
-    this.projectIngestSchema['properties']['content'] = this.projectMetadataSchema;
+    const queryParam = this.route.snapshot.queryParamMap;
+    this.projectFormData$ = of(this.emptyProject);
+
+    if (queryParam.has(Identifier.DOI)) {
+      this.fillProjectManually = false;
+      this.projectFormData$ = this.autofillProjectDetails(Identifier.DOI, queryParam.get(Identifier.DOI));
+    }
+
+    this.projectFormData$
+      .pipe(
+        concatMap(project => this.setSchema(project))
+      ).subscribe(
+      data => {
+        this.projectFormData = data;
```

Comment:
Not stricly neccesary but it makes it a bit cleaner. Instead of subscribing to the observable in the component class and writing data to another variable you can use Angular's `async` pipe in the template

Reply:
hmmm, I think we did that because we wanted the form to load even if there was an error in the observable. I wonder if there's a better way to do this 🤔

Reply:
yeah, there might be a way to catch and handle the error without subscribing to it, i also prefer the async pipe, just not sure how we can handle the error now...

Reply:
let's check later